test/unit/sdam/server_description.test.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from 'chai';
+
+import { Long, ObjectId } from '../../../src/bson';
+import {
+  compareTopologyVersion,
+  ServerDescription,
+  TopologyVersion
+} from '../../../src/sdam/server_description';
+
+describe('ServerDescription', function () {
+  describe('error equality', function () {
+    const errorEqualityTests = [
+      {
+        description: 'equal error types and messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: true
+      },
+      {
+        description: 'equal error types and unequal messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('blah') }),
+        equal: false
+      },
+      {
+        description: 'unequal error types and equal messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new TypeError('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: false
+      },
+      {
+        description: 'null lhs',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: null }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: false
+      },
+      {
+        description: 'null rhs',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new TypeError('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: undefined }),
+        equal: false
+      }
+    ];
+
+    for (const test of errorEqualityTests) {
+      it(test.description, function () {
+        expect(test.lhs.equals(test.rhs)).to.equal(test.equal);
+      });
+    }
+  });
+
+  it('should sensibly parse an ipv6 address', function () {
+    const description = new ServerDescription('[ABCD:f::abcd:abcd:abcd:abcd]:27017');
+    expect(description.host).to.equal('abcd:f::abcd:abcd:abcd:abcd');
+    expect(description.port).to.equal(27017);
+  });
+
+  describe('compareTopologyVersion()', () => {
+    const processIdZero = new ObjectId('00'.repeat(12));
+    const processIdOne = new ObjectId('00'.repeat(11) + '01');
+    const compareTopologyVersionTests: {
+      title: string;
+      currentTv?: TopologyVersion;
+      newTv?: TopologyVersion;
+      out: 0 | -1 | 1;
+    }[] = [
+      {
+        title: 'should return that currentTv and newTv are equal',
```

Comment:
can you fix the descriptions for all of these to actually state what condition is being tested?
e.g., in this case, `should return that the versions are equal  when process ids are equal and both counter values are Long.ZERO`, in the next case `should return that the versions are equal  when process ids are equal and both counter values are non-zero numbers` (and by the way, we should add plain number `0` as a test case)
I think this could also benefit  from splitting up the cases into several context blocks `when process ids are equal`, and `when process ids are not equal`

Reply:
Organized the tests, and improved names :)